spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/StreamDefinition.java

```diff
@@ -153,6 +149,28 @@ public void remove() {
 			throw new UnsupportedOperationException();
 		}
 
+		/**
+		 * Return whether or not this iterator is past the very first module to deploy.
+		 */
+		public boolean hasMoreDownstream() {
+			return i < modules.size() - 2;
```

Comment:
First of all, maybe we should add a check that this is called only once `next()` has been called at least once.

This is really about the position that is currently pointed to (ie the result of the last call to next). In your example, once next has been called once (yielding `log`), then there were no modules beyond that one. Remember those method names were for iterator consistency, but this is really `!isLast()`